### PR TITLE
Homework 2 task 5

### PR DESCRIPTION
**Pull request by ivanzuev78**
Homework 2 task 5 done

### CODE REVIEW

File: HomeWork_2/hw5.py [hunk 1]
```diff
@@ -14,3 +14,30 @@
 assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
 
 """
+from typing import List
+
+
+def custom_range(*args) -> List[str]:
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw5.py:20]: At least first argument is alway a sequence, so lets add it here.

File: HomeWork_2/hw5.py [hunk 2]
```diff
@@ -14,3 +14,30 @@
...
+
+    sequence, *params = args
+    step = 1
+    first_elem = 0
+    if isinstance(args[-1], int):
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw5.py:25]: What is this check about?

File: HomeWork_2/test_hw5.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import string
+
+from HomeWork_2.hw5 import custom_range
+
+
+def test_custom_range_example_1():
```
Comment (Cjkjvfnby) [inline HomeWork_2/test_hw5.py:6]: `any iterable of unique values` what does it means for you?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/hw5.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_2/hw5.py:25]: When the last argument is int, we use it as a step. So, I cheсk it.

File: HomeWork_2/test_hw5.py (hunk 3 above)
Comment (ivanzuev78) [reply HomeWork_2/test_hw5.py:6]: It means, that we have an object with the method __iter__, that returns inner objects, and all inner objects are unique.

File: HomeWork_2/hw5.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_2/hw5.py:20]: How we can do it? When we write "(*args: Sequence)" it means, that all args are Sequences, but the last one can be integer.
custom_range

File: HomeWork_2/test_hw5.py (hunk 3 above)
Comment (Cjkjvfnby) [reply HomeWork_2/test_hw5.py:6]: You havenot mention about objects type.  Are inner objects can be only strings or they can have any type? For example int?

File: HomeWork_2/hw5.py (hunk 1 above)
Comment (Cjkjvfnby) [reply HomeWork_2/hw5.py:20]: Positional arguments:  `def custom_range(sequence: Sequence, *args)`

Comment (Cjkjvfnby) [reply HomeWork_2/hw5.py:20]: `List[str]`  why?

File: HomeWork_2/hw5.py (hunk 2 above)
Comment (Cjkjvfnby) [reply HomeWork_2/hw5.py:25]: Can we check if step is specified in a diffeent way?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please see comment

File: HomeWork_2/hw5.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_2/hw5.py:20]: Because I didn't think about other types.

File: HomeWork_2/hw5.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_2/hw5.py:25]: Yes.

File: HomeWork_2/test_hw5.py (hunk 3 above)
Comment (ivanzuev78) [reply HomeWork_2/test_hw5.py:6]: Yes, I didn't think about types. This changes all.

File: HomeWork_2/hw5.py [hunk 4]
```diff
@@ -14,3 +14,29 @@
...
 """
+from typing import List, Sequence, Any
+
+
+def custom_range(sequence: Sequence, *args) -> List[Any]:
+
+    step = 1
+    first_elem = 0
+
+    if len(args) == 0:
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw5.py:25]: I think args should be at least size 1.  (stop | stop:start | stop:start:step )

File: HomeWork_2/hw5.py [hunk 5]
```diff
@@ -14,3 +14,29 @@
...
+    elif len(args) == 3:
+        if not isinstance(args[2], int):
+            raise TypeError("Step must be integer!")
+        last_elem = sequence.index(args[1])
+        first_elem = sequence.index(args[0])
+        step = args[2]
+    elif len(args) > 3:
+        raise IndexError("Too many arguments")
+
+    return [i for i in sequence[first_elem:last_elem:step]]
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw5.py:42]:
```
return sequence[first_elem:last_elem:step]
```

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**